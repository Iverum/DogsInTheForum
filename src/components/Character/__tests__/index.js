import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import { Character } from '../index'

jest.mock('firebase', () => {
  return {
    auth: () => {
      return {
        currentUser: { uid: 'testUser' },
        getRedirectResult: () => { return Promise.resolve(true) }
      }
    },
    database: () => {
      return {
        ref: () => {
          return {
            off: () => {},
            on: () => {}
          }
        }
      }
    }
  }
})

test('character is rendered correctly', () => {
  const component = renderer.create(
    <Character
      characters={[
        { name: 'Test Character' },
        { name: 'Test Character 2' }
      ]}
    />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
