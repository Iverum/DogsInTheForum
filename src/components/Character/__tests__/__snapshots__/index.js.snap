exports[`test character is rendered correctly 1`] = `
<div
  className="twelve columns">
  <header
    className="row">
    <h1
      className="nine columns">
      Characters
    </h1>
    <input
      className="button-primary three columns"
      onClick={[Function]}
      type="button"
      value="Create New Character" />
  </header>
  <div
    className="row">
    <div>
      <div
        className={undefined}>
        <div>
          <a
            className="button u-full-width"
            onClick={[Function]}
            style={Object {}}>
            Test Character
          </a>
        </div>
        <div>
          <a
            className="button u-full-width"
            onClick={[Function]}
            style={Object {}}>
            Test Character 2
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
